aipl/ops/llm.py

```diff
@@ -172,14 +172,25 @@ def embedding_openai(aipl, v:str, **kwargs) -> dict:
     return dict(model=kwargs.get('model'),
                 used_tokens=used,
                 embedding=resp['data'][0]['embedding'])
+ 
+def get_model_and_binary_paths() -> List[str]:
+    if 'MODELS_DIR' not in os.environ:
+        raise AIPLException('''MODELS_DIR envvar must be set for !llm-local''')
+    models_dir = Path(os.environ['MODELS_DIR'])
+    # binaries can be set per-model and globally, with preference for local
+    if os.path.isfile(models_dir/'main'):
+        inference_binary = models_dir/'main'
+    elif 'INFERENCE_BINARY' in os.environ:
```

Comment:
oh, oops--this code should be on the server side.

And to answer your question: my idea with INFERENCE_BINARY is that it's a catchall inference binary. Since some models need binaries specific to them, you'd put those in their folder and then that would take precedence. It's the best I've got so far, but will likely be reworking it again to support models that need python code to do inference on them.